funccount.py

```diff
@@ -0,0 +1,54 @@
+import os
+import re
+import signal
+import subprocess as sp
+import threading
+import time
+
+class Funccount:
+    def _sleep_and_run(self):
+        time.sleep(self.sleep)
+
+        cmd = [self.executable, self.funcs]
+
+        self.proc = sp.Popen(cmd, stdout=sp.PIPE, stderr=sp.PIPE, text=True,
+                             start_new_session=True)
+
+    def __init__(self, funcs, executable='funccount', sleep=0):
+        with open('/sys/kernel/debug/tracing/available_filter_functions') as f:
+            avail_funcs = f.read().splitlines()
+
+        for func in funcs.split():
+            r = re.compile(func.replace('*', '.*'))
+            if any(r.match(line) for line in avail_funcs):
+                continue
+            raise ValueError(f'Function/Regex not available for tracing: {func}')
+
+        self.funcs = funcs
+        self.executable = executable
+        self.sleep = sleep
+
+    def start(self):
+        self.thr = threading.Thread(target=self._sleep_and_run)
+        self.thr.start()
+
+    def stop(self):
+        self.thr.join()
+
+        pgid = os.getpgid(self.proc.pid)
+        os.killpg(pgid, signal.SIGINT)
+
+        while self.proc.poll() is None:
+            time.sleep(0.1)
+
+        out, err = self.proc.communicate()
+        if err:
```

Comment:
probably better to check self.proc.returncode than assume output to stderr is a failure.